starter-core/src/main/java/io/micronaut/starter/feature/build/gradle/Gradle.java

```diff
@@ -59,6 +61,22 @@ public void apply(GeneratorContext generatorContext) {
         generatorContext.addTemplate("gradleWrapper", new URLTemplate("gradlew", classLoader.getResource("gradle/gradlew"), true));
         generatorContext.addTemplate("gradleWrapperBat", new URLTemplate("gradlew.bat", classLoader.getResource("gradle/gradlew.bat"), true));
 
+        if (generatorContext.getFeatures().language().isKotlin() || generatorContext.getFeatures().testFramework().isKotlinTestFramework()) {
+            generatorContext.addBuildPlugin(GradlePlugin.builder().id("org.jetbrains.kotlin.jvm").lookupArtifactId("kotlin-gradle-plugin").build());
+            generatorContext.addBuildPlugin(GradlePlugin.builder().id("org.jetbrains.kotlin.kapt").lookupArtifactId("kotlin-gradle-plugin").build());
+            generatorContext.addBuildPlugin(GradlePlugin.builder().id("org.jetbrains.kotlin.plugin.allopen").lookupArtifactId("kotlin-allopen").build());
+            if (generatorContext.getFeatures().isFeaturePresent(JpaFeature.class)) {
+                generatorContext.addBuildPlugin(GradlePlugin.builder().id("org.jetbrains.kotlin.plugin.jpa").lookupArtifactId("kotlin-noarg").build());
+            }
+        }
+        if (generatorContext.getFeatures().language().isGroovy() || generatorContext.getFeatures().testFramework().isSpock()) {
+            generatorContext.addBuildPlugin(GradlePlugin.builder().id("groovy").build());
+        }
+        generatorContext.addBuildPlugin(GradlePlugin.builder().id(
+                (generatorContext.getFeatures().mainClass().isPresent() ||
+                        generatorContext.getFeatures().contains("oracle-function") ||
+                        generatorContext.getApplicationType() == ApplicationType.DEFAULT && generatorContext.getFeatures().contains("aws-lambda")) ?
+                "io.micronaut.application" : "io.micronaut.library").lookupArtifactId("micronaut-gradle-plugin").build());
```

Comment:
Can you please use an `if` or extract it to a private method or variable? It is difficult to read with the ternary operator inside the `.id(....)` method in the builder.

Reply:
I have extracted a method.